builtin/providers/azurerm/provider.go

```diff
@@ -0,0 +1,107 @@
+package azurerm
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+// Provider returns a terraform.ResourceProvider.
+func Provider() terraform.ResourceProvider {
+	return &schema.Provider{
+		Schema: map[string]*schema.Schema{
+			"arm_credentials": &schema.Schema{
+				Type:         schema.TypeString,
+				Optional:     true,
+				DefaultFunc:  schema.EnvDefaultFunc("ARM_CREDENTIALS", nil),
+				ValidateFunc: validateArmConfigFile,
+			},
+
+			"subscription_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("ARM_SUBSCRIPTION_ID", ""),
+			},
+
+			"client_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("ARM_CLIENT_ID", ""),
+			},
+
+			"client_secret": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("ARM_CLIENT_SECRET", ""),
+			},
+
+			"tenant_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("ARM_TENANT_ID", ""),
+			},
+		},
+
+		ResourcesMap: map[string]*schema.Resource{
+			"azurerm_resource_group":  resourceArmResourceGroup(),
+			"azurerm_virtual_network": resourceArmVirtualNetwork(),
+		},
+
+		ConfigureFunc: providerConfigure,
+	}
+}
+
+// Config is the configuration structure used to instantiate a
+// new Azure management client.
+type Config struct {
+	ManagementURL string
```

Comment:
ManagementURL is still not yet sourced or used anywhere, and I'm guessing it's just an 'artefact' of the copy of the config. Personally, considering that it will realistically be years until one would not want to use the default Management URL with ARM, I would drop it for now, maybe leave an open issue on it until the need pops up... (the fact that this even exists in the ARM API is to give people the option to create ASM-based resources through ARM on their own Azure Stack deployment (which noone will have a need for considering we have an ASM provider)).